File: 1200.minimum-absolute-difference.go
/*
 * @lc app=leetcode id=1200 lang=golang
 *
 * [1200] Minimum Absolute Difference
 */

package main

import (
	"math"
	"sort"
)

// @lc code=start
func minimumAbsDifference(arr []int) [][]int {
	sort.Ints(arr)

	diff := math.MaxInt
	for i := 0; i < len(arr)-1; i++ {
		diff = min(arr[i+1]-arr[i], diff)
	}

	var ret [][]int
	for i := 0; i < len(arr)-1; i++ {
		if diff == arr[i+1]-arr[i] {
			ret = append(ret, []int{arr[i], arr[i+1]})
		}
	}
	return ret
}

// @lc code=end

File: 1200.minimum-absolute-difference.cpp
/*
 * @lc app=leetcode id=1200 lang=cpp
 *
 * [1200] Minimum Absolute Difference
 */

#include <bits/stdc++.h>

using namespace std;

// @lc code=start
class Solution {
 public:
  vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
    sort(arr.begin(), arr.end());

    int diff = INT_MAX;
    for (int i = 0; i < arr.size() - 1; ++i)
      diff = min(diff, arr[i + 1] - arr[i]);

    vector<vector<int>> ret;
    for (int i = 0; i < arr.size() - 1; ++i)
      if (arr[i + 1] - arr[i] == diff) ret.push_back({arr[i], arr[i + 1]});

    return ret;
  }
};
// @lc code=end
